### DISCUSSION TITLE
Image uploads continuously failing

### DISCUSSION THREAD

**Discussion by melanierichards** [community/community#28947] [Other Feature Feedback, Questions, & Ideas] [answered]
Ironically I can't attach a screenshot to this discussion because I can't upload any issues to the Github comment box. Whether I use drag and drop or the file picker, the outcome is the same:

> Something went really wrong, and we can't process that file. Try again.

(Trying again doesn't help)

This seemed to be browser-specific at first but is preventing me from uploading any images across MS Edge, Chrome, and Safari (MacOS). I've tried things like disabling privacy/security features and extensions, and clearing browsing data (the latter two browsers I don't use day to day so should be pretty fresh anyway). Any pointers? Thanks!

**Comment by radar-blip** [comment]
Only thing I can think of is,  if your images dont have extensions like .jpg or .png it wont work. Other than that, I dont know.

**Reply by melanierichards** [reply]
Good point, should've clarified the extension for these issues is `.png`. It's a weird one, huh?

**Reply by radar-blip** [reply]
Yes, weird. 
I cant think of anything else it could be on account that im asleep. I hope you get it figured out.
Good luck!

**Reply by melanierichards** [reply]
Thank you, appreciate it! Have a good rest 😊

**Reply by radar-blip** [reply]
Hey, are all you images and extensions in lower-case? And have no spaces? I was reading some people were having problems with image files have spaces, and upper-case extensions like .JPG and .PNG. I'm sure you're smarter than all that though.

Anyway, If you can find no fix. You can always file a bug report here:
https://support.github.com/contact/bug-report  (if you like).

**Reply by melanierichards** [reply]
Yep, all that is good to go. Seems like this is crossing fully into bug report territory. Thanks for thinking of ideas with me!

**Comment by QASithLord** [comment]
> Ironically I can't attach a screenshot to this discussion because I can't upload any issues to the Github comment box. Whether I use drag and drop or the file picker, the outcome is the same:
> 
> > Something went really wrong, and we can't process that file. Try again.
> 
> (Trying again doesn't help)
> 
> This seemed to be browser-specific at first but is preventing me from uploading any images across MS Edge, Chrome, and Safari (MacOS). I've tried things like disabling privacy/security features and extensions, and clearing browsing data (the latter two browsers I don't use day to day so should be pretty fresh anyway). Any pointers? Thanks!

It is also failing for me on all uploads. Its been like this for a week now.

I turned on dev tools, and in the console I see this POST: attachment-upload.ts:114          POST https://github-production-user-asset-6210df.s3.amazonaws.com/ net::ERR_CERT_COMMON_NAME_INVALID

**Reply by melanierichards** [reply]
Same error for me! I'm going to submit a [bug report](https://support.github.com/contact/bug-report) and see if that helps.

**Reply by QASithLord** [reply]
I submitted a ticket. Hopefully they will figure this out. Its annoying to have to ask coworkers to upload images for me.

**Reply by QASithLord** [reply]
weird question : are you on the west coast/washington? making sure its not a location issue. My coworkers in the bay area are not having this issue.

**Reply by melanierichards** [reply]
Interesting. 🤔 I am indeed in Seattle, WA.

**Reply by QASithLord** [reply]
I logged out, cleared cache, restarted Safari and was able to post an image. However doing the same in Chrome failed.

**Reply by melanierichards** [reply]
I come bearing news/ideas! A coworker suggested tethering to my phone to see if changing my IP address makes a difference. That fixed the problem for me, though not a sustainable option. Something I noticed is that when tethered, I have both an IPv4 and IPv6 address. Untethered, I have just an IPv4 address. My coworkers who don't have the issue have both IPv4 and IPv6 addresses. So...I'm thinking potentially this resource expects an IPv6 address.

**Reply by melanierichards** [reply]
Setting up IPv6 doesn’t seem to have helped, maybe our IP range is flagged (we have Centurylink, maybe you have the same ISP?). I set up Private Relay in Safari and that helped fixed the issue, since it obfuscates the actual IP address.

**Edit:** just kidding, we just updated to using [Google's public DNS](https://developers.google.com/speed/public-dns/docs/using) (were using Cloudflare's) and it's now working without Private Relay.

**Reply by matt-augustine** [reply]
I'm in the Seattle area as well and have been experiencing this issue for the past few weeks. When I tried loading that ....s3.amazonaws.com URL in another browser tab, Chrome reported that the SSL cert was issued to *.safezone.mcafee.com and not amazonaws.com. This made me suspect that there was some security junk messing with DNS resolution or attempting to proxy requests.

Sure enough, I found this "Cyber Security" option had been enabled in the ZYXEL modem I got from my ISP, CenturyLink. Disabling it seems to have fixed the issue for me.
![image](https://user-images.githubusercontent.com/7634922/188002953-8a33c422-4515-4a77-8d21-df3aaa746c1f.png)

**Reply by agjohnson** [reply]
@matt-augustine wow, great find!

I'm in Spokane, also on CenturyLink, and still haven't switched off their supplied C4000XG router. I also had this option enabled by default, and after disabling, can't seem to reproduce this issue anymore :crossed_fingers:

**Reply by ebuhle** [reply]
> Sure enough, I found this "Cyber Security" option had been enabled in the ZYXEL modem I got from my ISP, CenturyLink. Disabling it seems to have fixed the issue for me. 

Another Seattle-area CenturyLink user here. This problem has been driving me crazy for a couple of months, and the above solution fixed it for me just as described. Thank you so much; I never would have figured this out on my own! I don't know as much about networking as some posters in this thread appear to, and I'm not sure about the implications of disabling a security feature on my modem (which I hadn't messed with recently, so I infer the change was on GitHub's end). But I have enough other layers of protection that I guess I won't worry about it if it means I can once again post images in GitHub Issues.

**Comment by melanierichards** [comment, marked as answer]
Solved my own issue:

1. Enabled IPv6 on my router.
2. Used [Google's public DNS](https://developers.google.com/speed/public-dns/docs/using) in System Preferences, replacing Cloudflare DNS.

AFAICT it was (2) that actually fixed the issue, vs (1).

**Reply by QASithLord** [reply]
Thanks for the info. But it would be weird. that I would have to do it. I've been having no problems for a year and a half and haven't changed anything.....

**Reply by melanierichards** [reply]
100% agreed there. I finally got a response on the private ticket and they seem to have fixed something on their end, so maybe try giving it a go?

**Reply by DanHickstein** [reply]
This also (quite unexpectedly!) worked for me. Mac users note that you need to fully exit the "Network" preferences panel (not just the "Advanced" panel) in order for the DNS changes to take place.

**Reply by geoffolson** [reply]
I did step 2 alone and that fixed it for me. Thank you.

**Reply by beedell-roke** [reply]
> I did step 2 alone and that fixed it for me.

Likewise, that [^1] remediated this for me:

[^1]: [`askubuntu.com/revisions/1546862/2#content`](https://askubuntu.com/revisions/1546862/2#content:~:text=Enter%20the%20IP%20addresses%20for,Screenshot)

![`{'Type'='Screenshot'; 'Date'='+20250501T001352+0100'}#`](https://github.com/user-attachments/assets/baaba106-bbab-4133-a79e-75e7cd170966) ![`{'Type'='Screenshot'; 'Date'='+20250501T001347+0100'}#`](https://github.com/user-attachments/assets/14297914-9fd7-4f4c-88fc-db3971c2c968)

Reported at [`support.github.com/ticket/personal/0/3367477#tc-37080149443860`](https://support.github.com/ticket/personal/0/3367477#tc-37080149443860:~:text=I%20am%20unable%20to%20upload%20images%20to%20an%20issue.).

**Comment by QASithLord** [comment]
Looks like it works! Huzzah. Wonder what they broke....

**Comment by matthewlein** [comment]
Wild, updating my DNS to google's also fixed it over here.

**Reply by jsjoeio** [reply]
Worked for me as well. Thank you!

**Comment by mihow** [comment]
Adding Google's DNS fixed the issue for me as well. 

![image](https://user-images.githubusercontent.com/158175/221989011-c590ba75-97e1-40a8-b98c-fc37dd68c761.png)

**Comment by leagris** [comment]
Is that all you have to suggest like: `Surrender your DNS queries to Google` as a solution to this still unresolved issue?

**Comment by peterflynn** [comment]
I hit this with an image file GitHub just didn't seem happy with – a basic 789 KB .jpg file, nothing unusual about it.  It failed every time both on my regular Internet and on corporate VPN, so it didn't seem like a DNS issue.

My solution was really simple: just take a screenshot of the image and paste that into the GitHub text editor instead 🙂  (I went through the clipboard but I bet dragging in a screenshot file would have worked too).

**Comment by ToshihitoKon** [comment]
I had the same issue where pasting images from clipboard didn’t work in Chrome, but it worked in Vivaldi.

In my case, the cause turned out to be the FoxyProxy Chrome extension. After disabling it, image pasting started working correctly in Chrome as well.

If you’re using FoxyProxy or a similar proxy-related extension, try disabling it and see if that resolves the issue.

Hope this helps someone!

**Comment by qwertychouskie** [comment]
Just started getting this issue.  No proxies or anything.  FF devtools show a CORS error.  Tried multiple DNS servers, no luck.

**Comment by darthdesi** [comment]
yup also dead here

**Comment by danielledeleo** [comment]
Happening for me too. Getting CORS 403s in the console.

**Reply by beedell-roke** [reply]
@danielledeleo, my HAR doesn't show `403`s, but I do see:

> ~~~JSON
> {
>   "name": "Sec-Fetch-Mode",
>   "value": "cors"
> }
> ~~~

...Relevant?

**Comment by drankinatty** [comment]
I have been unable to upload images to issues on github for week attempting results in (absolutely nothing) here. In issues I get an html comment that the image failed to upload. I have successfully uploaded issues for more than a decade before this, so something has changed, and nothing has changed on my end. I have always had a google dns as part of the forwarders list in my BIND9 setup:

```none
    forwarders { xxx.xxx.xxx.xxx; yyy.yyy.yyy.yyy; 8.8.4.4; };
````

I have not IPv6 and name resolution has not changed on my end. To the best of my recall, this image problem began March, April ?? 2026. Before that I had no issue loading images. Any other ideas?

**Reply by TheArgentinian** [reply]
Same here. Problems started about a couple of months ago. I use Adguard DNS and GitHub never had an issue with it. Something changed on their end .

**Comment by drankinatty** [comment]
At least for me I was able to track down the cause and find a solution. The problem for me was when you attempt to upload an image, github adds sites to handle the upload, e.g. `https://github-production-user-asset-6210df.s3.amazonaws.com/` which was not a site presented to you when you started the bug report or have had a chance to allow in your security settings, etc.. (I generally use `strict`) Using the Developer Tools in Firefox, I was presented with the following errors on upload:

```none
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://github-production-user-asset-6210df.s3.amazonaws.com/. (Reason: CORS request did not succeed). Status code: (null).

Uncaught AttachmentUploadError: App assets attachment upload failed: storage request.
    o attachment-upload.ts:63
    setTimeout handler*o attachment-upload.ts:62
    process attachment-upload.ts:104
    d batch-upload.ts:59
    819435 file-attachment.ts:36
    m index.js:96
    attach index.js:164
    p index.js:246
Caused by: error { isTrusted: true, lengthComputable: false, loaded: 0, total: 0, eventPhase: 0, bubbles: false, cancelable: false, returnValue: true, defaultPrevented: false, composed: false }
attachment-upload.ts:63:35

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://api.github.com/_private/browser/errors. (Reason: CORS request did not succeed). Status code: (null).
```

The problem was the `s3.amazonaws.com/` site which caused the CORS violation.  It does not show up as being part of the page until you attempt the upload. Only after you attempt an upload and it fails do you have the opportunity allow the additional sites by adjusting your security/blocker settings.
